### DISCUSSION TITLE
Including private data folder in build output

### DISCUSSION THREAD

**Discussion by Earl-Brown** [vercel/next.js#37347] [Help]
I'm building a small personal site, and rather than messing with database management, I'm using a folder with static files in it.

Then I have an API route which accesses those files during runtime.

Running on my development machine, this works just great.  When I build on Vercel, however, the folder is not included.

I've tried a number of approaches to make this data available, but haven't found anything that works.

At one point, I was able to include the content as part of the "public" folder, but that has two problems for me:
1 - the files are publicly exposed
2 - I don't have a way to search them

I fully admit I'm way out of my depth when it comes to webpack.

Is it possible to include a folder in the build output that isn't publicly exposed, but is visible to server functions?

**Comment by icyJoseph** [comment]
Hi,

The thing with api routes on production is that these are often run as tasks, so accessing files, and other resources included on the project, requires usage of `process.cwd()`, which will give you the path to where the server, and your files, are living.

Try to access those files by using resolve from `path`, sort of like, `path.resolve(process.cwd(), 'folder-name')`, and let me know how it goes.

**Comment by Earl-Brown** [comment]
I was able to do that - and a bunch of other experiments.  I even managed to list some of the files in my project.

But I couldn't get the folder I want to be included in what makes it to the "server".


That's the big problem I'm having.  I can get them there if I put them under "public" - but I'd like to save ideas on my site that I'm not ready to share with the world.  So having them in a public folder isn't the best of ideas. 






---- On Tue, 31 May 2022 23:22:06 -0700 Joseph ***@***.***> wrote ----





Hi,

The thing with api routes on production is that these are often run as tasks, so accessing them requires usage of process.cwd(), which will give you the path to where the server, and your files, are living.

Try to access those files by using resolve from path, sort of like, path.resolve(process.cwd(), 'folder-name'), and let me know how it goes.

**Reply by icyJoseph** [reply]
I guess it's time to pull out the big guns then, could you please read through here, https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions#next.js, basically root level api folders can use Vercel's serverless functions capability as well. For example, think of deploying a CRA to Vercel, there you could have a top level api folder that would work like pages/api does. 

The advantage is that you can tell Vercel which files to include when it builds!

**Comment by Earl-Brown** [comment]
Thank you!  That looks like EXACTLY what I was looking for!






---- On Wed, 01 Jun 2022 11:19:47 -0700 Joseph ***@***.***> wrote ----





I guess it's time to pull out the big guns then, could you please read through here, https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions#next.js, basically root level api folders can use Vercels capability as well. Think of deploying a CRA to Vercel, there you could have a top level api folder that would work like pages/api does.

The advantage is that you can tell Vercel which files to include when it builds!

**Reply by icyJoseph** [reply]
If it works out, please mark it as answered. A variation of this question comes up every now and then.

**Reply by Earl-Brown** [reply]
Done!

**Reply by Earl-Brown** [reply]
Sorry - undone.  

It turns out that link isn't the one I need.  It actually looks different from what I saw when I marked the answer as correct.

The link documents how to read files from the filesystem - I can't see anything in it that allows me to specify including a folder verbatim.

Like I said, though, I remember something different when I accepted this answer.

**Reply by icyJoseph** [reply]
They changed the documentation, I noticed that as well. They had an option to include files for API.
